internal/worker/task_handler.go

```diff
@@ -352,22 +375,32 @@ func removeFirmwareAlreadyAtDesiredVersion(hCtx *sm.HandlerContext, fws []*model
 
 		switch {
 		case !ok:
+			cause := "component not found in inventory"
 			hCtx.Logger.WithFields(logrus.Fields{
 				"component": fw.Component,
-			}).Warn("inventory missing component")
+			}).Warn(cause)
+
+			causes[fw.Component] = cause
```

Comment:
ack, I see your point here - but Im a bit confused how it relates to the referenced code.
Sorry if I'm missing the obvious

Reply:
Consider 2 firmware sets (A and B). A includes firmwares 1,2,3 while B has 1,2,5. Before we merge, trying to apply set A to a server that is expecting B (e.g. it doesn't have the component for firmware 3) fails. Now it will apply firmware 1 and 2 and the job succeeds. Is that the desired behavior? Is the user more surprised that his job failed from a data-input issue (they specified A instead of B) or more surprised that the job worked, but only applied firmwares 1 and 2 and left the last component alone?